platform/keysystem/keyclient/state/state.go

```diff
@@ -15,6 +16,29 @@ import (
 type ClientState struct {
 	Keyserver *server.Keyserver
 	Keygrant  *tls.Certificate
+
+	// entries are present here if an API was inaccessible because we weren't authorized for that particular API
+	// this lets us avoid constantly trying to request something from an API that isn't intended for us
+	RetryAt map[string]time.Time
+}
+
+func NewClientState(keyserver *server.Keyserver) *ClientState {
+	return &ClientState{
+		Keyserver: keyserver,
+		RetryAt:   make(map[string]time.Time),
+	}
+}
+
+// how frequently to recheck whether our access is denied, in case it changes
+const RetryAfter = time.Hour * 12
```

Comment:
Supposing we accidentally didn't grant enough permissions, is there an easy way we could manually force this to retry besides waiting 12 hours?  Would it suffice to restart whichever services were affected?

Reply:
Yes, retry timeouts aren't persisted to disk, so you can just roll the keyclient with `systemctl restart keyclient`. We already do this during the install process when we want it to pick up something immediately.